src/src/buffer_pool_manager/mod.rs

```diff
@@ -0,0 +1,22 @@
+use crate::shared::definitions::*;
+use buffer_pool::*;
+use std::collections::HashMap;
+
+mod buffer_pool;
+mod frame_buffer;
+mod page_table;
+
+#[derive(Default, Debug)]
+pub struct BufferPoolManager {
+    pools: HashMap<TableId, BufferPool>,
```

Comment:
This should be made available for concurrent access - we need multiple threads to have access to different pools